From 7b2e3b1b306f759c23a49d6007a8778b3ecd4fbf Mon Sep 17 00:00:00 2001
From: m34782 <lewis.hanly@microchip.com>
Date: Mon, 15 Jun 2020 08:35:52 +0100
Subject: [PATCH] PFSoC MAC Interface auto-negotiation.

Signed-off-by: Lewis Hanly <lewis.hanly@microchip.com>
---
 drivers/net/phy/vitesse.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/net/phy/vitesse.c b/drivers/net/phy/vitesse.c
index bb680352708a..9b2c5962d37b 100644
--- a/drivers/net/phy/vitesse.c
+++ b/drivers/net/phy/vitesse.c
@@ -99,6 +99,13 @@ static int vsc824x_add_skew(struct phy_device *phydev)
 static int vsc824x_config_init(struct phy_device *phydev)
 {
 	int err;
+        unsigned int val;
+
+        /* Enable MAC interface auto-negotiation */
+        phy_write(phydev, 0x1f, 0);
+        val = phy_read(phydev, 0x17);
+        val |= ((1 << 13) | 0x880);
+        phy_write(phydev, 0x17, val);
 
 	err = phy_write(phydev, MII_VSC8244_AUX_CONSTAT,
 			MII_VSC8244_AUXCONSTAT_INIT);
